src/main/java/com/epam/aidial/core/function/CollectAttachmentsFn.java

```diff
@@ -0,0 +1,76 @@
+package com.epam.aidial.core.function;
+
+import com.epam.aidial.core.Proxy;
+import com.epam.aidial.core.ProxyContext;
+import com.epam.aidial.core.config.ApiKeyData;
+import com.epam.aidial.core.security.AccessService;
+import com.epam.aidial.core.storage.ResourceDescription;
+import com.epam.aidial.core.util.HttpException;
+import com.epam.aidial.core.util.HttpStatus;
+import com.epam.aidial.core.util.ProxyUtil;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+import java.net.URI;
+
+@Slf4j
+public class CollectAttachmentsFn extends BaseFunction<ObjectNode> {
```

Comment:
please add a javadoc since this class not only collect attachments, but also assign per-request api-key

Reply:
fixed